File: baekjoon/StringQuestion/018-2935-소음.js
let [strA, opt, strB] = require('fs').readFileSync('z.in').toString().trim().split('\n');

let result, prefix;
if (opt == "+") {
    if (strA.length > strB.length) {
        prefix = strA.substring(0, (strA.length - strB.length));
        result = prefix + strB;
    }
    else if (strA.length == strB.length) {
        result = "2" + strA.substring(1, strA.length);
    }
    else {
        prefix = strB.substring(0, (strB.length - strA.length));
        result = prefix + strA;
    }
}
else {
    result = "1" + strA.substring(1, strA.length) + strB.substring(1, strB.length);
}

console.log(result);

File: baekjoon/StringQuestion/018-2935-소음.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();

        int lenA, lenB, opt;
        String strA, strB, plusPrefix;
        strA = sc.next();
        lenA = strA.length();
        opt = sc.next().charAt(0);
        strB = sc.next();
        lenB = strB.length();

        if (opt == '+') {
            if (lenA > lenB) {
                plusPrefix = strA.substring(0, (lenA - lenB));
                sb.append(plusPrefix).append(strB);
            }
            else if (lenA == lenB) {
                sb.append("2").append(strA.substring(1, lenA));
            }
            else {
                plusPrefix = strB.substring(0, (lenB - lenA));
                sb.append(plusPrefix).append(strA);
            }
        }
        else {
            sb.append("1").append(strA.substring(1, lenA)).append(strB.substring(1, lenB));
        }   

        System.out.print(sb);
    }
}